src/Templates/WorkflowTemplate.php

```diff
@@ -92,29 +94,29 @@ public function __construct($name, $description = '', $version = '0.0', $remindD
         $this->sort = $sort;
     }
 
-    public function getName()
+    public function getName(): string
     {
-        return $this->name;
+        return (string) $this->name;
     }
 
-    public function getVersion()
+    public function getVersion(): string
     {
-        return $this->version;
+        return (string) $this->version;
     }
 
-    public function getDescription()
+    public function getDescription(): string
     {
-        return $this->description;
+        return (string) $this->description;
     }
 
-    public function getRemindDays()
+    public function getRemindDays(): int
     {
-        return $this->remindDays;
+        return (int) $this->remindDays;
     }
 
-    public function getSort()
+    public function getSort(): int
     {
-        return $this->sort;
+        return (int) $this->sort;
     }
```

Comment:
Why are we adding strong typing here, but not to the properties? Backwards compatibility doesn't seem like a sensible reason on its own, because strong typing is explicitly a breaking change - as as I mentioned above, YAML type inference is very much a thing.

This class isn't a `DataObject` and doesn't directly interact with `DBField`, so I'm struggling to see how this is related at all to the issue.

Reply:
As mentioned in the comment above, the properties need to be dynamic because it's likely that the types used in YAML are the wrong type, though we don't want people things to break.  However we do need to cast to the correct type e.g. DBVarchar will fail validation if an int or float is returned for getVersion()